From 9b22dd34a39b396f6cdf89c5025c6912587d3a3d Mon Sep 17 00:00:00 2001
From: Chrissi^ <chris@tinyhost.de>
Date: Sun, 1 Sep 2019 13:09:33 +0200
Subject: [PATCH 08/38] radvd: announce IPv6 of VPN-node as DNS-Server

Using this change an VPN-node announces it's own IPv6 as DNS server.
/tmp/addr6 is written by nodeoute.lua during runtime configuration.

Signed-off-by: Chrissi^ <chris@tinyhost.de>
---
 .../luasrc/lib/gluon/radvd/arguments                 | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/package/gluon-mesh-batman-adv/luasrc/lib/gluon/radvd/arguments b/package/gluon-mesh-batman-adv/luasrc/lib/gluon/radvd/arguments
index fd214c29..d450b81e 100755
--- a/package/gluon-mesh-batman-adv/luasrc/lib/gluon/radvd/arguments
+++ b/package/gluon-mesh-batman-adv/luasrc/lib/gluon/radvd/arguments
@@ -3,12 +3,14 @@
 local site = require "gluon.site"
 
 local f = io.open("/tmp/range6","r")
-if f then
+local g = io.open("/tmp/addr6", "r")
+
+if f and g then
     local range6 = f:read('*a')
+    local addr6  = g:read('*a')
     f:close()
-    io.write("-i br-client -p " .. range6)
+    g:close()
 
-    if site.dns.servers() and site.next_node.ip6() then
-        io.write(" --rdnss " .. site.next_node.ip6())
-    end
+    io.write("-i br-client -p " .. range6)
+    io.write(" --rdnss " .. addr6)
 end
-- 
2.34.1

